File: 01_JAVA/52_EX_Stream/src/com/lion/model/BasketBallStudentClass.java
package com.lion.model;

import com.lion.util.Values;

import java.util.Scanner;

public class BasketBallStudentClass extends StudentClass{
    // 리바운드 개수
    int reboundCount;

    @Override
    public void inputStudentInfo(Scanner scanner) {
        type = Values.TYPE_BASKETBALL;
        System.out.println();
        System.out.println("농구부 학생 정보 입력");
        super.inputStudentInfo(scanner);
        System.out.print("리바운드 개수 : ");
        reboundCount = scanner.nextInt();
    }

    @Override
    public void showStudentInfo() {
        System.out.println();
        System.out.println("소속 운동부 : 농구부");
        super.showStudentInfo();
        System.out.printf("리바운드 개수 : %d개\n", reboundCount);
    }

    @Override
    public void showAction() {
        System.out.println("농구 선수가 블로킹을 합니다");
    }
}

File: 02_Kotlin/20_EX/src/com/lion/school/BasketBallStudentClass.kt
package com.lion.school

import java.util.*

class BasketBallStudentClass(var reboundCount:Int = 0) : StudentClass() {

    // 학생 정보를 입력받는 메서드
    override fun inputStudentInfo(scanner: Scanner) {
        println()
        println("농구부 학생 정보를 입력해주세요")
        super.inputStudentInfo(scanner)
        print("리바운드 개수 : ")
        reboundCount = scanner.nextInt()
    }

    // 학생 정보를 출력하는 메서드
    override fun printStudentInfo() {
        println()
        println("농구부 학생 정보")
        super.printStudentInfo()
        println("리바운드 개수 : ${reboundCount}개")
    }

    // 달리는 메서드
    override fun run() {
        println("농구 선수가 달립니다")
    }

    // 블록킹 하는 기능
    fun blocking(){
        println("블록킹을 합니다")
    }
}